### PR TITLE
Inspector bars

### PR DESCRIPTION
**Pull request by vpereverzev**
Peek 2020-06-30 23-07
Peek 2020-06-30 23-09
Peek 2020-07-02 19-15
Peek 2020-07-02 19-16

### CODE REVIEW

**Comment by Jojo-Schmitz** [comment]
"Bar" rather then "Measure"? Too much British influence?

File: inspectors/models/inspectorlistmodel.cpp [hunk 1]
```diff
@@ -7,6 +7,7 @@
 #include "text/textsettingsmodel.h"
 #include "score/scoredisplaysettingsmodel.h"
 #include "score/scoreappearancesettingsmodel.h"
+#include "score/barssettingsmodel.h"
```
Comment (Howard-C) [inline inspectors/models/inspectorlistmodel.cpp:10]: Should be `barsettingsmodel.h`.

File: inspectors/models/score/barssettingsmodel.cpp [hunk 2]
```diff
@@ -0,0 +1,74 @@
+#include "barssettingsmodel.h"
+
+BarsSettingsModel::BarsSettingsModel(QObject* parent, IElementRepositoryService* repository) :
```
Comment (Howard-C) [inline inspectors/models/score/barssettingsmodel.cpp:3]: Here and all others too, `BarSettingsModel`.

Comment (Jojo-Schmitz) [reply inspectors/models/score/barssettingsmodel.cpp:3]: or `MeasureSettingsModel`

File: inspectors/models/inspectorlistmodel.cpp (hunk 1 above)
Comment (Jojo-Schmitz) [reply inspectors/models/inspectorlistmodel.cpp:10]: Or `measuresettingsmodel.h`

File: inspectors/inspectorssetup.cpp [hunk 3]
```diff
@@ -60,6 +60,7 @@ void InspectorsSetup::registerResources()
 #include "types/scoreappearancetypes.h"
 #include "types/bendtypes.h"
 #include "types/tremolobartypes.h"
+#include "types/bartypes.h"
```
Comment (Jojo-Schmitz) [inline inspectors/inspectorssetup.cpp:63]: `measuretypes.h`

File: inspectors/inspectorssetup.cpp [hunk 4]
```diff
@@ -108,4 +109,5 @@ void InspectorsSetup::registerUiTypes()
     qmlRegisterType<Ms::GridCanvas>("MuseScore.Inspectors", 3, 3, "GridCanvas");
     qmlRegisterUncreatableType<BendTypes>("MuseScore.Inspectors", 3, 3, "BendTypes", "Not creatable as it is an enum type");
     qmlRegisterUncreatableType<TremoloBarTypes>("MuseScore.Inspectors", 3, 3, "TremoloBarTypes", "Not creatable as it is an enum type");
+    qmlRegisterUncreatableType<BarTypes>("MuseScore.Inspectors", 3, 3, "BarTypes", "Not creatable as it is an enum type");
```
Comment (Jojo-Schmitz) [inline inspectors/inspectorssetup.cpp:112]: `MeasureTypes`

File: inspectors/models/abstractinspectormodel.cpp [hunk 5]
```diff
@@ -52,6 +52,12 @@ static const QList<Ms::ElementType> TEXT_ELEMENT_TYPES = {
     Ms::ElementType::SYSTEM_TEXT
 };
 
+static const QList<Ms::ElementType> BAR_ELEMENT_TYPES = {
```
Comment (Jojo-Schmitz) [inline inspectors/models/abstractinspectormodel.cpp:55]: `MEASURE_ELEMENT_TYPES`

File: inspectors/models/abstractinspectormodel.cpp [hunk 6]
```diff
@@ -92,6 +98,8 @@ AbstractInspectorModel::InspectorSectionType AbstractInspectorModel::sectionType
         return SECTION_NOTATION;
     } else if (TEXT_ELEMENT_TYPES.contains(elementType)) {
         return SECTION_TEXT;
+    } else if (elementType == Ms::ElementType::MEASURE) {
+        return SECTION_BAR;
```
Comment (Jojo-Schmitz) [inline inspectors/models/abstractinspectormodel.cpp:102]: `SECTION_MEASURE`

File: inspectors/models/abstractinspectormodel.cpp [hunk 7]
```diff
@@ -114,6 +122,9 @@ QList<Ms::ElementType> AbstractInspectorModel::supportedElementTypesBySectionTyp
     case SECTION_TEXT: {
         return TEXT_ELEMENT_TYPES;
     }
+    case SECTION_BAR: {
```
Comment (Jojo-Schmitz) [inline inspectors/models/abstractinspectormodel.cpp:125]: dito

File: inspectors/models/abstractinspectormodel.h [hunk 8]
```diff
@@ -26,6 +26,7 @@ class AbstractInspectorModel : public QObject
 public:
     enum InspectorSectionType {
         SECTION_UNDEFINED = -1,
+        SECTION_BAR,
```
Comment (Jojo-Schmitz) [inline inspectors/models/abstractinspectormodel.h:29]: dito

File: inspectors/models/inspectorlistmodel.cpp [hunk 9]
```diff
@@ -102,6 +106,9 @@ void InspectorListModel::createModelsBySectionType(const QList<AbstractInspector
         beginInsertRows(QModelIndex(), rowCount(), rowCount());
 
         switch (modelType) {
+        case SectionType::SECTION_BAR:
```
Comment (Jojo-Schmitz) [inline inspectors/models/inspectorlistmodel.cpp:109]: dito

File: inspectors/models/inspectorlistmodel.cpp [hunk 10]
```diff
@@ -102,6 +106,9 @@ void InspectorListModel::createModelsBySectionType(const QList<AbstractInspector
...
+            m_modelList << new BarsSettingsModel(this, m_repository);
```
Comment (Jojo-Schmitz) [inline inspectors/models/inspectorlistmodel.cpp:110]: `MeasureSettingsModel`

File: inspectors/models/score/barssettingsmodel.cpp [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+    AbstractInspectorModel(parent, repository)
+{
+    setTitle(tr("Bars"));
+    setSectionType(SECTION_BAR);
+}
+
+void BarsSettingsModel::insertBars()
```
Comment (Jojo-Schmitz) [inline inspectors/models/score/barssettingsmodel.cpp:10]: `MeasureSettingsModel::inserMeasures`

Comment (Howard-C) [reply inspectors/models/score/barssettingsmodel.cpp:10]: `inser**t**Measures()`

Comment (Howard-C) [reply inspectors/models/score/barssettingsmodel.cpp:10]: `insertMeasures()`

**Comment by vpereverzev** [comment]
> "Bar" rather then "Measure"? Too much British influence?

hah, yes, I think I can agree with your comments

**Comment by Jojo-Schmitz** [comment]
Why has it been closed? See also https://musescore.org/en/node/314047, where something similar is requested for MuseScore 3 (I assume this here to have been meant for MuseScore 4)

**Pull request closed without merging**